source/components/NumericInput.js

```diff
@@ -0,0 +1,224 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { isFunction, isString, flow, pick } from 'lodash';
+import Input from './Input';
+import FormField from './FormField';
+
+export default class NumericInput extends FormField {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      caretPosition: 0,
+      separatorsCounter: 0,
```

Comment:
Please add some doc comments for the props … for me it's not totally clear what a `separatorsCounter` is 😉